### PR TITLE
Add `sent_at` to envelope header item

### PR DESCRIPTION
**Pull request by denrase**
## :scroll: Description

- Add `sent_at` property to envelope header
- Set to current date before passing envelope to transport

## :bulb: Motivation and Context

Relates to https://github.com/getsentry/sentry-dart/issues/1337

Doc: https://develop.sentry.dev/sdk/envelopes/

## :green_heart: How did you test it?

Unit tests

## :pencil: Checklist

- [x] I reviewed the submitted code.
- [x] I added tests to verify the changes.
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled.
- [ ] I updated the docs if needed.
- [ ] Review from the native team if needed.
- [x] No breaking change or entry added to the changelog.
- [x] No breaking change for hybrid SDKs or communicated to hybrid SDKs.

## :crystal_ball: Next steps

Add to cocoa sdk

### CODE REVIEW

File: sentry/src/main/java/io/sentry/SentryClient.java [hunk 1]
```diff
@@ -514,6 +518,8 @@ public void captureSession(final @NotNull Session session, final @Nullable Hint
 
     try {
       hint.clear();
+      SentryDate now = options.getDateProvider().now();
```
Comment (markushi) [inline sentry/src/main/java/io/sentry/SentryClient.java:521]:
```suggestion
      final @NotNull SentryDate now = options.getDateProvider().now();
```

**Review by markushi — COMMENTED** [review]
Looking good! I left a few comments for some minor improvements

File: sentry/src/main/java/io/sentry/SentryEnvelopeHeader.java [hunk 2]
```diff
@@ -110,6 +126,12 @@ public static final class Deserializer implements JsonDeserializer<SentryEnvelop
           case JsonKeys.TRACE:
             traceContext = reader.nextOrNull(logger, new TraceContext.Deserializer());
             break;
+          case JsonKeys.SENT_AT:
+            String dateValue = reader.nextStringOrNull();
+            if (dateValue != null) {
+              sentAt = DateUtils.getDateTime(dateValue);
+            }
```
[the remark this answers is not in the document]
Comment (denrase) [reply sentry/src/main/java/io/sentry/SentryEnvelopeHeader.java]: Ah, probably should have remember **that** API. ^^

**Comment by denrase** [comment]
Changed the implementation due to the discussion on the cocoa PR

https://github.com/getsentry/sentry-cocoa/pull/2859

**Review by romtsn — APPROVED** [review]

**Comment by romtsn** [comment]
thanks @denrase 👍

**Pull request merged**